src/interfaces/swig/SGBase.i

```diff
@@ -485,6 +491,33 @@ namespace shogun
         }
 
         /*int getbuffer(PyObject *obj, Py_buffer *view, int flags) { return 0; }*/
+
+        std::vector<std::string> parameter_names() const {
+			std::vector<std::string> result;
+			for (auto const& each: self->get_parameters()) {
+				result.emplace_back(each.first);
+			}
+			return result;
+        }
+
+		PyObject* parameter_types() const {
+			PyObject* py_result = PyDict_New();
```

Comment:
I’m unsure about this....
We would want simple c++ code that is then exposed to swig which does the mapping. In principle.
Maybe we should rather work on getting the typemaps to work?

Reply:
Also much less code....

Reply:
I tried to template std::map<std::string, std::string> but SWIG didn't return a dictionary... Which I thought it would?

Reply:
It depends on whether swig actually has typemaps for maps. And I also remember that sometimes swig was a diva with matching those, eg we had to move stuff to shogun.i as otherwise it didn’t match (some order problem)
Even if there are no typemaps for map I think it would be better to just write those ourselves and keep the code general here.

Reply:
You mean return a map<string, string> here and then write a %typemap(argout) for each target language that can't translate map<string, string> to a builtin data structure?

Reply:
So I had a better look into SWIG's std::map wrapper, and it behaves very similarly to a dict (i.e. has an item getter like behaviour, key and values getter). It just looks weird when printed, which could be odd for a user:
```python
print(kernel.parameter_types())
<Swig Object of type 'std::map< std::string,std::string,std::less< std::string >,std::allocator< std::pair< std::string const,std::string > > > *' at 0x1181de490>
```
So should probably customise the map<string, string> typemap right (at least for python)?

Reply:
one option. The printed thing looks ugly indeed